### PR TITLE
Improve the documentation for AS OF SYSTEM TIME.

### PR DESCRIPTION
**Pull request by knz**
Following up to  #3018.

Now that there's a separate page, it should be the central place to
get information about the feature. Concentrate the information there.

@jseldess if you're OK with this change, and after you suggest the necessary edits, I'll volunteer to propagate the change to the other version directories.

### CODE REVIEW

File: v2.1/as-of-system-time.md [hunk 1]
```diff
@@ -43,6 +112,52 @@ Assuming the following statements are run at `2016-01-01 12:00:00`, they would e
...
+at least the top-level statement. It is not valid to use it only in a
+[subquery](subqueries.html).
+
+For example, the following is invalid:
+
+~~~
+SELECT * FROM (SELECT * FROM t AS OF SYSTEM TIME '-4h'), u
+~~~
+
+To facilitate the composition of larger queries from simpler queries,
```
Comment (jseldess) [inline v2.1/as-of-system-time.md]: nit: remove `will` and change `allow` to `allows`

File: v2.1/as-of-system-time.md [hunk 2]
```diff
@@ -43,6 +112,52 @@ Assuming the following statements are run at `2016-01-01 12:00:00`, they would e
...
+
+### Using `AS OF SYSTEM TIME` In Subqueries
+
+To enable time travel, the `AS OF SYSTEM TIME` clause must appear in
+at least the top-level statement. It is not valid to use it only in a
+[subquery](subqueries.html).
+
+For example, the following is invalid:
+
+~~~
```
Comment (jseldess) [inline v2.1/as-of-system-time.md]: Add `sql` syntax highlighting, the `>` prompt, and copy-to-clipboard.

File: v2.1/select-clause.md [hunk 3]
```diff
@@ -394,91 +394,11 @@ HAVING COUNT(name) > 1;
...
-    WHERE name = (SELECT name FROM customers AS OF SYSTEM TIME '2018-04-05 16:00:00'
-      WHERE name ~ 'Vain' LIMIT 1);
-~~~
-
-~~~
-ERROR:  cannot specify AS OF SYSTEM TIME with different timestamps
-~~~
+CockroachDB lets you find data as it was stored at a given point in
+time using `AS OF SYSTEM TIME` with various [supported
+formats](as-of-system-time.html).  This can be also advantageous for
```
Comment (jseldess) [inline v2.1/select-clause.md]: nit: use only one space after a period.

**Comment by knz** [comment]
Review status: 0 of 3 files reviewed at latest revision, 11 unresolved discussions.

---

*[v2.1/as-of-system-time.md, line 17 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-17:-LCnlHPKBOPhq0aNOsJh:b19609c) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L17)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'm not sure why, but the line breaks in this callout are causing problems with markdown>html conversion. On the test site, everything under ## Synopsis is broken.

Can you make this a single line? I tested it locally, and it fixes the problem.
I just tested 
</blockquote></details>

This is a bug in the converter. Have you filed this bug already?

---

*[v2.1/as-of-system-time.md, line 22 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-22:-LCnlLxzEHIdbelV1QQu:b-r5uqmn) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L22)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Let's put `<div id="toc"></div>` here.
</blockquote></details>

Are you asking me to do this are you suggesting we do it together?

---

*[v2.1/as-of-system-time.md, line 29 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-29:-LCnlSlKD3ZPAY-BH4nv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L29)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`parameters to` > `parameters of the`
</blockquote></details>

Done.

---

*[v2.1/as-of-system-time.md, line 38 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-38:-LCnlTVY9CNsk86wuNxv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L38)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: 
> The `timestamp` argument supports the following formats:
</blockquote></details>

Done.

---

*[v2.1/as-of-system-time.md, line 115 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-115:-LCnlZJe3Rj6lAi--2EY:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L115)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `From` > `from`
</blockquote></details>

Done.

---

*[v2.1/as-of-system-time.md, line 117 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-117:-LCnlhcI9nZaTSoMyOF4:bilbc6r) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L117)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think I'd make this a callout after the code blocks. Also, I'd rephrase the second sentence a bit:

> It is not yet possible to select from multiple tables at different timestamps. The entire query runs at the specified time in the past. 
</blockquote></details>

I am reluctant to use a callout because then it does not show up in the TOC.

Otherwise rephrased.

---

*[v2.1/as-of-system-time.md, line 127 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-127:-LCnlp642UlGtU4-sXHv:b-x0bhqm) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L127)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Let's break these into 3 separate code blocks, each with copy-to-clipboard.
</blockquote></details>

Are you asking me to do this are you suggesting we do it together? I have no idea.

---

*[v2.1/as-of-system-time.md, line 132 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-132:-LCnlxYV58LmiO1flKTZ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L132)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `In` > `in`
</blockquote></details>

Done.

---

*[v2.1/as-of-system-time.md, line 140 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-140:-LCnlylC2lj_pjp_Xisg:boxu2hq) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L140)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Add `sql` syntax highlighting, the `>` prompt, and copy-to-clipboard.
</blockquote></details>

This does not make sense - the syntax is invalid. There is nothing to try out.

---

*[v2.1/as-of-system-time.md, line 144 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-144:-LCnm47JDQ2fz17dj_07:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L144)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: remove `will` and change `allow` to `allows`
</blockquote></details>

Done.

---

*[v2.1/select-clause.md, line 399 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJA-r1-399:-LCnmA6G8gfmO7IJPLJ_:bx9dkgs) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/select-clause.md#L399)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: use only one space after a period.
</blockquote></details>

This is markup. This space does not exist in the generated HTML.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3155)*

**Comment by jseldess** [comment]
:lgtm:

---

Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions, all commit checks successful.

---

*[v2.1/as-of-system-time.md, line 17 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-17:-LD1sxZ4-bc5_IOyo4nA:b5dzn8t) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L17)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

This is a bug in the converter. Have you filed this bug already?
</blockquote></details>

I haven't files this as a bug in our repo. Do you mean this is a bug in redcarpet or on our end?

---

*[v2.1/as-of-system-time.md, line 22 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-22:-LD1t7ub-83ies8_iVeK:bdthinq) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L22)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Are you asking me to do this are you suggesting we do it together?
</blockquote></details>

Sorry to have been unclear. I was asking you to do it.

---

*[v2.1/as-of-system-time.md, line 117 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-117:-LD1tPjVBCL3lWopvkDi:b-6plsn5) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L117)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I am reluctant to use a callout because then it does not show up in the TOC.

Otherwise rephrased.

</blockquote></details>

Oh, I wasn't suggesting making this entire section a callout, just this first paragraph. But it's good as-is.

---

*[v2.1/as-of-system-time.md, line 127 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-127:-LD1tWUB4K9Ffek-t8LZ:b-32sub4) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L127)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Are you asking me to do this are you suggesting we do it together? I have no idea.
</blockquote></details>

Again, sorry for the confusion here. When I say "let's" in cases like this, I'm really asking you to do it but on behalf of "us". I'll be more direct in the future. 

---

*[v2.1/select-clause.md, line 399 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJA-r1-399:-LD1tzGDBHt8cI4Wzx6B:b-rebab6) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/select-clause.md#L399)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

This is markup. This space does not exist in the generated HTML.
</blockquote></details>

True. Fine by me.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3155#-:-LD1u6y83YYr7pnQByQK:bnfp4nl)*

**Comment by knz** [comment]
I have copied the changes to the v1.1 and v2.0 trees. RFAL.

---

Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[v2.1/as-of-system-time.md, line 17 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-17:-LD1wDr70wySfUuB06Kj:b-g0ztfr) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L17)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I haven't files this as a bug in our repo. Do you mean this is a bug in redcarpet or on our end?
</blockquote></details>

I do not know what "redcarpet" is. I do see however this is a bug in the jekyll-markdown translation.

---

*[v2.1/as-of-system-time.md, line 117 at r1](https://reviewable.io/reviews/cockroachdb/docs/3155#-LChTG_JKmL8gBeiXVJ8-r1-117:-LD1wVDzB8Qgw5tULOa7:b-9foege) ([raw file](https://github.com/cockroachdb/docs/blob/13bc6b6a37fa7d67e68cae97a0bac8a2cc8595de/v2.1/as-of-system-time.md#L117)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Oh, I wasn't suggesting making this entire section a callout, just this first paragraph. But it's good as-is.
</blockquote></details>

Added the callout.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3155#-:-LD1yAmN1TWcGI1Nhe-b:bdbl3on)*
